abstract_domains/usage/stack.py

```diff
@@ -72,13 +73,14 @@ def enter_if(self):
         return self
 
     def exit_if(self):
-        if self.is_bottom():
-            return self
-        self.pop()
+        self._postponed_pushpop.append(self._postponed_exit_if)
         return self
 
     def _output(self, output: Expression) -> 'UsedStack':
-        return self  # nothing to be done
+        if self.is_bottom():
+            return self
+        self.stack[-1]._output(output)
```

Comment:
Why not call `output()` instead (passing a set containing the output expression)?